File: Payhub - SDK/php_SDK/com/payhub/ws/vt/Batch.php
<?php
/**
 * Created by PhpStorm.
 * User: agustin
 * Date: 07/19/16
 * Time: 3:02 PM
 */

class Batch
{
    public static $BATCH_LINK = "batch/";
    private $terminalId;
    private $MSG;
    private $orgnazationId;
    private $batchId;
    private $TRN_RESPONSE;

    public function __construct()
    {
    }

    public static function fromArray($data)
    {
        $settings = new Batch();
        foreach ($data as $key => $value) {
            if (property_exists(get_class($settings), $key)) {
                if ($key == "TRN_RESPONSE") {
                    $_tmp = TransactionResponse::fromArray($value);
                    $settings->{$key} = $_tmp;
                }else{
                    $settings->{$key} = $value;
                }

            }
        }
        return $settings;
    }

    /**
     * @return mixed
     */
    public function getTerminalId()
    {
        return $this->terminalId;
    }

    /**
     * @param mixed $terminalId
     */
    public function setTerminalId($terminalId)
    {
        $this->terminalId = $terminalId;
    }

    /**
     * @return mixed
     */
    public function getMSG()
    {
        return $this->MSG;
    }

    /**
     * @param mixed $MSG
     */
    public function setMSG($MSG)
    {
        $this->MSG = $MSG;
    }

    /**
     * @return mixed
     */
    public function getOrgnazationId()
    {
        return $this->orgnazationId;
    }

    /**
     * @param mixed $orgnazationId
     */
    public function setOrgnazationId($orgnazationId)
    {
        $this->orgnazationId = $orgnazationId;
    }

    /**
     * @return mixed
     */
    public function getBatchId()
    {
        return $this->batchId;
    }

    /**
     * @param mixed $batchId
     */
    public function setBatchId($batchId)
    {
        $this->batchId = $batchId;
    }

    /**
     * @return mixed
     */
    public function getTRNRESPONSE()
    {
        return $this->TRN_RESPONSE;
    }

    /**
     * @param mixed $TRN_RESPONSE
     */
    public function setTRNRESPONSE($TRN_RESPONSE)
    {
        $this->TRN_RESPONSE = $TRN_RESPONSE;
    }


}
class TransactionResponse{
    private $RESPONSE_TEXT;
    private $RESPONSE_CODE;
    private $TRN_DATE_TIME;

    public function __construct()
    {
    }
    public static function fromArray($data)
    {

        $settings = new TransactionResponse();
        foreach ($data as $key => $value) {
            if (property_exists(get_class($settings), $key)) {
                $settings->{$key} = $value;
            }
        }
        return $settings;
    }

    /**
     * @return mixed
     */
    public function getRESPONSETEXT()
    {
        return $this->RESPONSE_TEXT;
    }

    /**
     * @param mixed $RESPONSE_TEXT
     */
    public function setRESPONSETEXT($RESPONSE_TEXT)
    {
        $this->RESPONSE_TEXT = $RESPONSE_TEXT;
    }

    /**
     * @return mixed
     */
    public function getRESPONSECODE()
    {
        return $this->RESPONSE_CODE;
    }

    /**
     * @param mixed $RESPONSE_CODE
     */
    public function setRESPONSECODE($RESPONSE_CODE)
    {
        $this->RESPONSE_CODE = $RESPONSE_CODE;
    }

    /**
     * @return mixed
     */
    public function getTRNDATETIME()
    {
        return $this->TRN_DATE_TIME;
    }

    /**
     * @param mixed $TRN_DATE_TIME
     */
    public function setTRNDATETIME($TRN_DATE_TIME)
    {
        $this->TRN_DATE_TIME = $TRN_DATE_TIME;
    }

}

File: Payhub - SDK/java_SDK/PayHubSDK/src/main/java/com/payhub/ws/vt/Batch.java
package com.payhub.ws.vt;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.payhub.ws.api.Errors;

@JsonInclude(JsonInclude.Include.NON_NULL)  
public class Batch{
	public static final String BATCH_LINK = "batch/";

	
	
	
	private String terminalId;
	@JsonProperty("MSG")
	private String msg;
	private String orgnazationId;
	private String batchId;
	@JsonProperty("TRN_RESPONSE")
	private TransactionResponse trnResponse;
	private List<Errors> errors;
	
	
	public static class TransactionResponse{
		@JsonProperty("RESPONSE_TEXT")		
		private String responseText;
		@JsonProperty("RESPONSE_CODE")
		private String responseCode;
		@JsonProperty("TRN_DATE_TIME")
		private String trnDateTime;
		/**
		 * @return the responseText
		 */
		public String getResponseText() {
			return responseText;
		}
		/**
		 * @param responseText the responseText to set
		 */
		
		public void setResponseText(String responseText) {
			this.responseText = responseText;
		}
		/**
		 * @return the responseCode
		 */
		public String getResponseCode() {
			return responseCode;
		}
		/**
		 * @param responseCode the responseCode to set
		 */
		
		public void setResponseCode(String responseCode) {
			this.responseCode = responseCode;
		}
		/**
		 * @return the trnDateTime
		 */
		public String getTrnDateTime() {
			return trnDateTime;
		}
		/**
		 * @param trnDateTime the trnDateTime to set
		 */
		
		public void setTrnDateTime(String trnDateTime) {
			this.trnDateTime = trnDateTime;
		}
		
	}


	/**
	 * @return the terminalId
	 */
	public String getTerminalId() {
		return terminalId;
	}


	/**
	 * @param terminalId the terminalId to set
	 */
	public void setTerminalId(String terminalId) {
		this.terminalId = terminalId;
	}


	/**
	 * @return the msg
	 */
	public String getMsg() {
		return msg;
	}


	/**
	 * @param msg the msg to set
	 */	
	public void setMsg(String msg) {
		this.msg = msg;
	}


	/**
	 * @return the orgnazationId
	 */
	public String getOrgnazationId() {
		return orgnazationId;
	}


	/**
	 * @param orgnazationId the orgnazationId to set
	 */
	public void setOrgnazationId(String orgnazationId) {
		this.orgnazationId = orgnazationId;
	}


	/**
	 * @return the batchId
	 */
	public String getBatchId() {
		return batchId;
	}


	/**
	 * @param batchId the batchId to set
	 */
	public void setBatchId(String batchId) {
		this.batchId = batchId;
	}


	/**
	 * @return the trnResponse
	 */
	public TransactionResponse getTrnResponse() {
		return trnResponse;
	}


	/**
	 * @param trnResponse the trnResponse to set
	 */
	public void setTrnResponse(TransactionResponse trnResponse) {
		this.trnResponse = trnResponse;
	}


	/**
	 * @return the errors
	 */
	public List<Errors> getErrors() {
		return errors;
	}


	/**
	 * @param errors the errors to set
	 */
	public void setErrors(List<Errors> errors) {
		this.errors = errors;
	}

    
}
